session/schema_amender.go

```diff
@@ -200,6 +200,13 @@ func (a *amendCollector) collectModifyColAmendOps(tblAtStart, tblAtCommit table.
 			if err != nil {
 				return nil, err
 			}
+		} else {
+			// If the column could not be found in the original schema, it could not be decided if this column
+			// is newly added or modified from an original column.Report error to solve the issue
+			// https://github.com/pingcap/tidb/issues/21470. This change will make amend fail for adding column
+			// and modifying columns at the same time.
+			return nil, errors.Trace(errors.Errorf("column=%v id=%v is not found for table=%v checking column modify",
```

Comment:
Is it ok to make add-column amend fail?

Reply:
Single add-column will not fail, change type containing add column with modify/change column will.

Reply:
It seems that `errors.Errorf` will attach stack info, thus I think `errors.Trace` is unnecessary.

Reply:
Looks like the function's comment at L193 needs to be  updated?